server/lib/realtime/helpers.ex

```diff
@@ -0,0 +1,25 @@
+defmodule Realtime.Helpers do
+  # key1=value1:key2=value2 to [{"key1", "value1"},{"key2", "value2"}]
+  @spec env_kv_to_list(String.t() | nil, keyword() | []) :: {:ok, keyword()} | :error
+  def env_kv_to_list("", _), do: :error
+
+  def env_kv_to_list(env_val, def_list) when is_binary(env_val) do
+    keywords =
+      String.split(env_val, ":")
+      |> Enum.map(
+        &(String.split(&1, "=")
+          |> List.to_tuple())
+      )
+      |> Enum.concat(def_list)
+
+    try do
+      # check if keywords is valid
+      Keyword.values(keywords)
+      {:ok, keywords}
+    rescue
+      _ -> :error
+    end
```

Comment:
@abc3 if `def_list` is always a keyword list of `[{"Content-Type", "application/json"}]`, since we're hardcoding it, is this check necessary?

```elixir
String.split(env_val, ":")
|> Enum.map(
  &(String.split(&1, "=")
    |> List.to_tuple())
)
```

will always return a keyword list. But if you think it is then I suggest `Keyword.keyword?/1`.

Reply:
Not exactly, 
the string "key1=value1key2=value2" (missing separator ":" between value1 and key2) return `[{"key1", "value1key2", "value2"}]` which is not two-element tuple.

Anyway, I've already rewritten this function. So, check the update, please.